### ISSUE TITLE
Command line refactor

### ISSUE THREAD

**Issue by grantat** [oduwsdl/CarbonDate#18]
In relation to a discussion in issue #15, command line arguments have proven to be problematic in the server version. I think it has warranted a rewrite of how command line arguments are parsed as well as how the server and local version are actually executed in [main.py](https://github.com/oduwsdl/CarbonDate/blob/master/main.py#L37).

**Comment by ibnesayeed** [comment]
Yes please! My OCD kicks in badly every time I look at the code.
